### PR TITLE
v4: Async LDAP connection fixes

### PR DESCRIPTION
**Pull request by ndptech**
A series of small fixes / changes to the async LDAP connection code.

### CODE REVIEW

File: src/lib/ldap/bind.c [hunk 1]
```diff
@@ -220,7 +220,7 @@ int fr_ldap_bind_async(fr_ldap_connection_t *c,
 
 	el = c->conn->el;
 
-	if (ldap_get_option(c->handle, LDAP_OPT_DESC, &fd) == LDAP_SUCCESS) {
+	if ((ldap_get_option(c->handle, LDAP_OPT_DESC, &fd) == LDAP_SUCCESS) && (fd >= 0)) {
```
Comment (arr2036) [inline src/lib/ldap/bind.c:222]: Could you add a comment here to.  That's very surprising behaviour, but given the quality of the libldap code it doesn't surprise me.

File: src/lib/ldap/bind.c [hunk 2]
```diff
@@ -123,12 +123,6 @@ static void _ldap_bind_io_write(fr_event_list_t *el, int fd, UNUSED int flags, v
 			      NUM_ELEMENTS(our_clientctrls),
 			      c, bind_ctx->serverctrls, bind_ctx->clientctrls);
 
-	/*
-	 *	Set timeout to be 0.0, which is the magic
-	 *	non-blocking value.
-	 */
-	(void) ldap_set_option(c->handle, LDAP_OPT_NETWORK_TIMEOUT, &fr_time_delta_to_timeval(0));
-
```
Comment (arr2036) [inline src/lib/ldap/bind.c:130]: Mention this in the other comment

File: src/lib/ldap/connection.c [hunk 3]
```diff
@@ -436,6 +436,7 @@ static fr_connection_state_t _ldap_connection_init(void **h, fr_connection_t *co
 	fr_ldap_state_t		state;
 
 	c = fr_ldap_connection_alloc(conn);
+	c->conn = conn;
```
Comment (arr2036) [inline src/lib/ldap/connection.c:439]: Is this for signalling purposes?

Comment (ndptech) [reply src/lib/ldap/connection.c:439]: It's for having access to the event list in fr_ldap_bind_async()

**Pull request merged**